contentcuration/contentcuration/views.py

```diff
@@ -204,10 +206,11 @@ def duplicate_nodes(request):
         nodes = node_ids.split()
         new_nodes = []
 
-        for node_id in nodes:
-            new_node = _duplicate_node(node_id, sort_order=sort_order, parent=target_parent)
-            new_nodes.append(new_node.pk)
-            sort_order+=1
+        with transaction.atomic():
```

Comment:
It was mostly an attempt to solve the timeout issue with the larger channels not duplicating fast enough. However, it seems there needs to be more done to avoid the 502/504 errors